posthog/views/message_preferences.py

```diff
@@ -0,0 +1,73 @@
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+from django.views.decorators.csrf import csrf_protect
+from django.views.decorators.http import require_http_methods
+from django.http import JsonResponse
+import structlog
+
+from posthog.models.message_preferences import RecipientIdentifier, MessageCategory, MessagePreference
+
+logger = structlog.get_logger(__name__)
+
+
+@require_http_methods(["GET"])
+def preferences_page(request: HttpRequest, token: str) -> HttpResponse:
```

Comment:
This is essentially our API so I'd really like some kind of tests for this if its the direction we are going in.

Alternatively I wonder if its worth a little more effort to have an actual API with a dedicated frontend that calls it instead 🤔